nixos/modules/services/databases/redis.nix

```diff
@@ -277,11 +283,18 @@ in
 
       serviceConfig = {
         ExecStart = "${cfg.package}/bin/redis-server /run/redis/redis.conf";
-        RuntimeDirectory = "redis";
-        StateDirectory = "redis";
         Type = "notify";
+        # User and group
         User = "redis";
         Group = "redis";
+        # Runtime directory and mode
+        RuntimeDirectory = "redis";
+        RuntimeDirectoryMode = "0750";
```

Comment:
*Maybe* we want to make these directives conditional:
```
serviceConfig = mkMerge [
  { User = ...;
    Group = ...;
  }
  (mkIf (cfg.settings == "/var/lib/redis") {
    StateDirectory = "redis";
    StateDirectoryMode = "0750";
  })
];
```

Reply:
Not workin this variant. Parameter `StateDirectory` does not apply.

Reply:
The `dir` parameter cannot be changed.
With:
```
services.redis.settings = { dir = "/var/lib/redis-example"; };
```
Error:
```
building all machine configurations...
error: The option `services.redis.settings.dir' has conflicting definition values:
- In `/home/.../mastodon.nix': "/var/lib/redis-example"
- In `/home/.../nixos/modules/services/databases/redis.nix': "/var/lib/redis"

```

Reply:
You need `lib.mkForce` to change the value.

Reply:
With:
```
services.redis.settings = lib.mkForce { dir = "/var/lib/redis-example"; };
```
Is writted such a configuration file:
```
cat /run/redis/redis.conf
```
```
dir /var/lib/redis-example
```
Result - redis not working.